api/v1alpha1/nodehealthcheck_types.go

```diff
@@ -118,6 +132,39 @@ type UnhealthyCondition struct {
 	Duration metav1.Duration `json:"duration"`
 }
 
+// EscalatingRemediation defines a remediation template with order and timeout
+type EscalatingRemediation struct {
+	// RemediationTemplate is a reference to a remediation template
+	// provided by a remediation provider.
+	//
+	// If a node needs remediation the controller will create an object from this template
+	// and then it should be picked up by a remediation provider.
+	//
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	RemediationTemplate corev1.ObjectReference `json:"remediationTemplate"`
+
+	// Order defines the order for this remediation.
+	// Remediations with lower order will be used before remediations with higher order.
+	// Remediations must not have the same order.
+	//
+	//+operator-sdk:csv:customresourcedefinitions:type=spec
+	Order int `json:"order"`
```

Comment:
Nit: do we want to allow negative values ?

Reply:
Good question, but I don't see a reason to not allow them. The definition is clear, and it doesn't matter if negative values are used: "Remediations with lower order will be used before remediations with higher order."